ext/mysql2/violite.h

```diff
@@ -0,0 +1,280 @@
+/* Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights
```

Comment:
Ok, that's a great lead. Let's see if there's a way to determine the connection type. It might be nice to expose that information, too like `Mysql2::Client#ssl_connection?`